## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 513528`
Commit: `477e06fc1c9918ae1a65b71083c7fa8236f34497`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>fromager</li>
    <li>fromager.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python313Packages.mailbits</li>
    <li>python313Packages.mailbits.dist</li>
    <li>python313Packages.pypi-simple</li>
    <li>python313Packages.pypi-simple.dist</li>
    <li>python314Packages.mailbits</li>
    <li>python314Packages.mailbits.dist</li>
    <li>python314Packages.pypi-simple</li>
    <li>python314Packages.pypi-simple.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>fromager</li>
    <li>fromager.dist</li>
    <li>python313Packages.mailbits</li>
    <li>python313Packages.mailbits.dist</li>
    <li>python313Packages.pypi-simple</li>
    <li>python313Packages.pypi-simple.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python314Packages.mailbits</li>
    <li>python314Packages.mailbits.dist</li>
    <li>python314Packages.pypi-simple</li>
    <li>python314Packages.pypi-simple.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>fromager</li>
    <li>fromager.dist</li>
    <li>python313Packages.mailbits</li>
    <li>python313Packages.mailbits.dist</li>
    <li>python313Packages.pypi-simple</li>
    <li>python313Packages.pypi-simple.dist</li>
    <li>python314Packages.mailbits</li>
    <li>python314Packages.mailbits.dist</li>
    <li>python314Packages.pypi-simple</li>
    <li>python314Packages.pypi-simple.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>fromager</summary>
<pre>  Caused by: dns error
  Caused by: failed to lookup address information: Temporary failure in name resolution
DEBUG Transient request failure for https://pypi.org/simple/hatchling/, retrying: error sending request for url (https://pypi.org/simple/hatchling/)
  Caused by: client error (Connect)
  Caused by: dns error
  Caused by: failed to lookup address information: Temporary failure in name resolution
DEBUG Transient request failure for https://pypi.org/simple/hatch-vcs/, retrying: error sending request for url (https://pypi.org/simple/hatch-vcs/)
  Caused by: client error (Connect)
  Caused by: dns error
  Caused by: failed to lookup address information: Temporary failure in name resolution
error: Request failed after 3 retries in 6.2s
  Caused by: Failed to fetch: `https://pypi.org/simple/hatch-vcs/`
  Caused by: error sending request for url (https://pypi.org/simple/hatch-vcs/)
  Caused by: client error (Connect)
  Caused by: dns error
  Caused by: failed to lookup address information: Temporary failure in name resolution
=========================== short test summary info ============================
FAILED tests/test_dependencies.py::test_get_build_backend_dependencies - subprocess.CalledProcessError: Command &#x27;[&#x27;uv&#x27;, &#x27;pip&#x27;, &#x27;install&#x27;, &#x27;--verbose...
FAILED tests/test_dependencies.py::test_get_build_sdist_dependencies - subprocess.CalledProcessError: Command &#x27;[&#x27;uv&#x27;, &#x27;pip&#x27;, &#x27;install&#x27;, &#x27;--verbose...
================== 2 failed, 492 passed, 2 skipped in 15.49s ===================</pre>
</details>
</details>

--- a/pkgs/by-name/fr/fromager/package.nix
+++ b/pkgs/by-name/fr/fromager/package.nix
@@ -1,53 +1,54 @@
 {
   lib,
-  stdenv,
-  python3,
   fetchFromGitHub,
+  python3Packages,
   writableTmpDirAsHomeHook,
 }:
 
-python3.pkgs.buildPythonApplication (finalAttrs: {
+python3Packages.buildPythonApplication (finalAttrs: {
   pname = "fromager";
-  version = "0.71.0";
+  version = "0.82.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "python-wheel-build";
     repo = "fromager";
     tag = finalAttrs.version;
-    hash = "sha256-3zz37BZx8FcKNl8mSmClIrZxvL+2AS0hJDct6K7BhBE=";
+    hash = "sha256-RlmbpnnwOWM5KLo4z1brdZnULGk3raGYassEqwWy0W0=";
   };
 
-  build-system = with python3.pkgs; [
+  build-system = with python3Packages; [
     hatchling
     hatch-vcs
   ];
 
-  dependencies = with python3.pkgs; [
+  dependencies = with python3Packages; [
     click
     elfdeps
-    html5lib
+    license-expression
     packaging
-    pkginfo
+    packageurl-python
     psutil
     pydantic
+    pypi-simple
     pyproject-hooks
     pyyaml
     requests
-    requests-mock
     resolvelib
     rich
-    setuptools
+    starlette
     stevedore
     tomlkit
     tqdm
     uv
+    uvicorn
     wheel
   ];
 
-  nativeCheckInputs = with python3.pkgs; [
+  nativeCheckInputs = with python3Packages; [
     pytestCheckHook
     requests-mock
+    spdx-tools
     twine
     uv
     writableTmpDirAsHomeHook
@@ -57,20 +58,16 @@ python3.pkgs.buildPythonApplication (finalAttrs: {
     "fromager"
   ];
 
-  disabledTestPaths = [
-    # Depends on wheel.cli module that is private since wheel 0.46.0
-    "tests/test_wheels.py"
-  ];
+  # Upstream runs pytest with `--log-level DEBUG`, which this test suite
+  # relies on for caplog assertions against INFO records.
+  # Reported: https://github.com/python-wheel-build/fromager/issues/1092
+  pytestFlags = [ "--log-level=DEBUG" ];
 
   disabledTests = [
     # Accessing pypi.org (not allowed in sandbox)
+    # Fixed in: https://github.com/python-wheel-build/fromager/pull/1093
     "test_get_build_backend_dependencies"
     "test_get_build_sdist_dependencies"
-  ]
-  ++ lib.optionals (stdenv.hostPlatform.isAarch64 && stdenv.hostPlatform.isLinux) [
-    # Assumes platform.machine() returns 'arm64' on ARM64, which is not true for Linux.
-    # Re-enable once https://github.com/python-wheel-build/fromager/pull/849 is merged.
-    "test_add_constraint_conflict"
   ];
 
   meta = {

